go/libraries/doltcore/sqle/stats/stats_provider.go

```diff
@@ -317,10 +456,14 @@ func (p *Provider) RowCount(ctx *sql.Context, db, table string) (uint64, error)
 
 func (p *Provider) DataLength(_ *sql.Context, db, table string) (uint64, error) {
 	var avgSize uint64
-	for meta, s := range p.dbStats[strings.ToLower(db)].stats {
-		if strings.EqualFold(db, meta.Database) && strings.EqualFold(table, meta.Table()) {
-			if s.AvgSize > avgSize {
-				avgSize = s.AvgSize
+	if dbStat := p.getStats(strings.ToLower(db)); dbStat != nil {
+		dbStat.mu.Lock()
+		defer dbStat.mu.Unlock()
+		for qual, s := range dbStat.stats {
+			if strings.EqualFold(db, qual.Database) && strings.EqualFold(table, qual.Table()) {
+				if s.AvgSize > avgSize {
+					avgSize = s.AvgSize
```

Comment:
Why do we take the highest value for avgSize across all stats?

Reply:
Changed this to match the primary index